node/typescript/v2/AmpliApp/src/ampli/index.ts

```diff
@@ -0,0 +1,906 @@
+  event_type = 'EventMaxIntForTest';
+
+  constructor(
+    public event_properties: EventMaxIntForTestProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventNoProperties implements BaseEvent {
+  event_type = 'Event No Properties';
+}
+
+export class EventObjectTypes implements BaseEvent {
+  event_type = 'Event Object Types';
+
+  constructor(
+    public event_properties: EventObjectTypesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventWithAllProperties implements BaseEvent {
+  event_type = 'Event With All Properties';
+  event_properties: EventWithAllPropertiesProperties & {
+    'requiredConst': "some-const-value";
+  };
+
+  constructor(
+    event_properties: EventWithAllPropertiesProperties,
+  ) {
+    this.event_properties = {
+      ...event_properties,
+      'requiredConst': "some-const-value",
+    };
+  }
+}
+
+export class EventWithArrayTypes implements BaseEvent {
+  event_type = 'Event With Array Types';
+
+  constructor(
+    public event_properties: EventWithArrayTypesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventWithConstTypes implements BaseEvent {
+  event_type = 'Event With Const Types';
+  event_properties = {
+    'Boolean Const': true,
+    'Integer Const': 10,
+    'Number Const': 2.2,
+    'String Const': "String-Constant",
+    'String Const WIth Quotes': "\"String \"Const With\" Quotes\"",
+    'String Int Const': 0,
+  };
+}
+
+export class EventWithDifferentCasingTypes implements BaseEvent {
+  event_type = 'event withDifferent_CasingTypes';
+
+  constructor(
+    public event_properties: EventWithDifferentCasingTypesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventWithEnumTypes implements BaseEvent {
+  event_type = 'Event With Enum Types';
+
+  constructor(
+    public event_properties: EventWithEnumTypesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventWithOptionalArrayTypes implements BaseEvent {
+  event_type = 'Event With Optional Array Types';
+
+  constructor(
+    public event_properties?: EventWithOptionalArrayTypesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventWithOptionalProperties implements BaseEvent {
+  event_type = 'Event With Optional Properties';
+
+  constructor(
+    public event_properties?: EventWithOptionalPropertiesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class EventWithTemplateProperties implements BaseEvent {
+  event_type = 'Event With Template Properties';
+
+  constructor(
+    public event_properties: EventWithTemplatePropertiesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export type PromiseResult<T> = { promise: Promise<T | void> };
+
+const getVoidPromiseResult = () => ({ promise: Promise.resolve() });
```

Comment:
Can you please move this to the core SDK?

We could add PromiseReturn to https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-types/src/amplitude-promise.ts
And getVoidPromiseReturn to https://github.com/amplitude/Amplitude-TypeScript/blob/7d11fdbee2ece65b64cec873d2f2ab74ed5fecf3/packages/analytics-core/src/utils/return-wrapper.ts

Reply:
That types are not used in the core SDK. Should we move them?

Reply:
Hmm good question. I guess for now let's leave it in codegen.